developer/services/qrexec.md

```diff
@@ -0,0 +1,307 @@
+
+    $ qrexec-client -d someVM user:bash
+
+The `qvm-run` command is heavily based on `qrexec-client`.
+It also takes care of additional activities, e.g. starting the domain if it is not up yet and starting the GUI daemon.
+Thus, it is usually more convenient to use `qvm-run`.
+
+There can be almost arbitrary number of `qrexec-client` processes for a domain (so, connected to the same `qrexec-daemon`, same domain) -- their data is multiplexed independently.
+Number of available vchan channels is the limiting factor here, it depends on the underlying hypervisor.
+
+## Qubes RPC services
+
+Some common tasks (like copying files between VMs) have an RPC-like structure: a process in one VM (say, the file sender) needs to invoke and send/receive data to some process in other VM (say, the file receiver).
+The Qubes RPC framework was created to securely facilite a range of such actions.
+
+Obviously, inter-VM communication must be tightly controlled to prevent one VM from taking control of another, possibly more privileged, VM.
+Therefore the design decision was made to pass all control communication via dom0, that can enforce proper authorization.
+Then, it is natural to reuse the already-existing qrexec framework.
+
+Also, note that bare qrexec provides `VM <-> dom0` connectivity, but the command execution is always initiated by dom0.
+There are cases when VM needs to invoke and send data to a command in dom0 (e.g. to pass information on newly installed `.desktop` files).
+Thus, the framework allows dom0 to be the RPC target as well.
+
+Thanks to the framework, RPC programs are very simple -- both RPC client and server just use their stdin/stdout to pass data.
+The framework does all the inner work to connect these processes to each other via `qrexec-daemon` and `qrexec-agent`.
+Additionally, disposable VMs are tightly integrated -- RPC to a DisposableVM is identical to RPC to a normal domain, all one needs is to pass `$dispvm` as the remote domain name.
+
+## Qubes RPC administration
+
+<!-- (*TODO: fix for non-linux dom0*) -->
+
+The dom0 directory `/etc/qubes-rpc/policy/` contains files for each available RPC action.
+Together their contents make up the RPC access policy database.
+Currently, the defined actions are:
+
+    qubes.ClipboardPaste
+    qubes.Filecopy
+    qubes.GetImageRGBA
+    qubes.GetRandomizedTime
+    qubes.Gpg
+    qubes.GpgImportKey
+    qubes.InputKeyboard
+    qubes.InputMouse
+    qubes.NotifyTools
+    qubes.NotifyUpdates
+    qubes.OpenInVM
+    qubes.OpenURL
+    qubes.PdfConvert
+    qubes.ReceiveUpdates
+    qubes.SyncAppMenus
+    qubes.USB
+    qubes.VMShell
+    qubes.WindowIconUpdater
+
+These files contain lines with the following format:
+
+    srcvm destvm (allow|deny|ask)[,user=user_to_run_as][,target=VM_to_redirect_to]
+
+You can specify srcvm and destvm by name or by one of three reserved keywords: `@anyvm`, `@dispvm`, and `dom0` (without the `@`).
```

Comment:
Actually, there is also `@` keyword for dom0: `@adminvm`

Reply:
There are more keywords allowed here, as described below: `@tag:...`, `@type:...`.